Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift

```diff
@@ -0,0 +1,141 @@
+// swiftlint:disable force_try
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupportTesting
+import XCTest
+
+@testable import TuistKit
+@testable import TuistSupport
+
+public enum Destination {
+    case simulator, device
+}
+
+open class TuistAcceptanceTestCase: TuistTestCase {
+    public var xcodeprojPath: AbsolutePath!
+    public var workspacePath: AbsolutePath!
+    public var fixturePath: AbsolutePath!
+    public var derivedDataPath: AbsolutePath!
+    public var cacheDirectory: AbsolutePath!
+
+    private var sourceRootPath: AbsolutePath!
+
+    override open func setUp() {
+        super.setUp()
+
+        cacheDirectory = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        derivedDataPath = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        setenv(
+            Constants.EnvironmentVariables.forceConfigCacheDirectory,
+            cacheDirectory.pathString,
+            1
+        )
+
+        sourceRootPath = try! AbsolutePath(
+            validating: ProcessInfo.processInfo.environment[
+                "TUIST_CONFIG_SRCROOT"
+            ]!
+        )
+        environment.tuistConfigVariables[
+            Constants.EnvironmentVariables.xcbeautifyBinaryPath
+        ] = sourceRootPath
+            .appending(components: ["vendor", ".build", "debug", "xcbeautify"])
+            .pathString
+
+        DeveloperEnvironment.shared = DeveloperEnvironment()
```

Comment:
This is another example of another mutable state that's shared across parallel tests. I'd revisit the approach.

Reply:
As mentioned above, there's no parallel state.

Reply:
I might be missing something, but I [don't think tests are isolated](https://github.com/tuist/tuist/pull/5553#discussion_r1392419939) at the environment variables level.